File: backend/GameApp.Application/UseCases/GameUseCases/GameCreateUseCase.cs
using GameApp.Domain.Entities;
using GameApp.Domain.Entities.Scenes;
using GameApp.Domain.Enumerates;
using GameApp.Domain.ValueObjects.Characters;

namespace GameApp.Application.UseCases.GameUseCases;

public interface GameCreateUseCase
{
    public Task<Game?> CreateGame(GameDifficulty difficulty, Character character, int numberScenesToFinish, FinalScene finalScene, List<Scene> listCurrentScenes, List<UserAction> listCurrentUserActions);
}

File: frontend/gameapp-frontend/src/application/usecases/game-use-cases/game-create-use-case.ts
import { Character } from '../../../domain/value-objects/characters/character';
import { Scene } from '../../../domain/entities/scenes/scene';
import type { Game } from '../../../domain/entities/game';
import type { FinalScene } from '../../../domain/entities/scenes/final-scene';
import type { GameDifficulty } from '../../../domain/enumerates/game-difficulty';

export interface IGameCreateUseCase {
  createGame(
    difficulty: GameDifficulty,
    character: Character,
    numberScenesToFinish: number,
    finalScene: FinalScene,
    listCurrentScenes: Scene[],
    listCurrentUserActions: string[]
  ): Promise<Game>;
}
